vignettes/sir-model.Rmd

```diff
@@ -173,4 +173,142 @@ ggplot(combined_solutions, aes(x = time, y = value, colour = name, linetype = ty
+
+# Comparison to other age matrices
+
+Now that we've established an age-structured SIR model, we can repeat the process with `conmat` matrices. This process is the same as in the Data Sources vignette.
+
+```{r setup-aus-matrices}
+world_data <- socialmixr::wpp_age() %>%
+    mutate(
+        new_lower_age = if_else(lower.age.limit >= 80, 80L, lower.age.limit)
+    ) %>%
+    group_by(new_lower_age, country, year) %>%
+    summarise(
+        population = sum(population)
+    )
+germany_2015 <- age_population(
+    data = world_data,
+    location_col = country,
+    location = "Germany",
+    age_col = new_lower_age,
+    year_col = year,
+    year = 2015
+)
+
+germany_2015
+```
+
+```{r conmat-germany}
+age_breaks_socialmixr <- c(seq(0, 75, by = 5), Inf)
+germany_contacts <- extrapolate_polymod(
+    population = germany_2015,
+    age_breaks = age_breaks_socialmixr
+)
+
+transmission_matrix <- get_setting_transmission_matrices(
+    age_breaks = age_breaks_socialmixr
+)
+
+n_finite_states <- length(age_breaks_socialmixr) - 1
+transmission_matrix$home <- matrix(4, nrow = n_finite_states, ncol = n_finite_states)
+transmission_matrix$work <- matrix(4, nrow = n_finite_states, ncol = n_finite_states)
+transmission_matrix$school <- matrix(4, nrow = n_finite_states, ncol = n_finite_states)
+transmission_matrix$other <- matrix(4, nrow = n_finite_states, ncol = n_finite_states)
+
+parameters <- list(
+    "transmission_matrix" = transmission_matrix,
+    "homogeneous_contact" = germany_contacts,
+    "gamma" = 1,
+    "s_indexes" = 1:n_finite_states,
+    "i_indexes" = (n_finite_states + 1):(2 * n_finite_states),
+    "r_indexes" = (2 * n_finite_states + 1):(3 * n_finite_states)
+)
+
+S0 <- rep(999, times = n_finite_states)
+I0 <- rep(1, times = n_finite_states)
+R0 <- rep(0, times = n_finite_states)
+initial_condition <- c(S0, I0, R0)
+names(initial_condition) <- paste(rep(c("S0", "I0", "R0"), each = n_finite_states), age_breaks_socialmixr[1:n_finite_states], sep = "_")
+```
+
+```{r solve-ode-germany}
+times <- seq(0, 100, by = 0.1)
+germany_soln <- ode(y = initial_condition, times = times, func = agestructured_sir, parms = parameters)
+
+# Have to convert ode output to a data frame to do any plotting
+germany_soln <- as.data.frame(germany_soln)
+
+tail(germany_soln)
+```
+
+```{r germany-sir-plot}
+germany_soln %>%
+    pivot_longer(!time) %>%
+    mutate(parent_state = substr(name, 1, 1)) %>%
+    group_by(time, parent_state) %>%
+    summarise(value = sum(value)) %>%
+    ungroup() %>%
+    rename(name = parent_state) %>%
+    mutate(type = "Age structured") %>%
+    ggplot(aes(x = time, y = value / sum(initial_condition), colour = name)) +
+    geom_line() +
+    labs(x = "Time", y = "Proportion")
+```
+
+Let's compare to the Prem matrices. Prem only has 16 age classes so we do need to re-do our population.
+```{r load-prem}
+str(prem_germany_contact_matrices)
```

Comment:
@njtierney can you think of a cleaner way to print these matrices? I could maybe cast them to the `conmat_setting_prediction_matrix` but I'm not sure if that will work.